create example-node (#16)
w3c/vc-jose-cose-test-suite: implementations/example-node/README.md

> This is a example Node implementation CLI wrapper for an implementation. Once the CLI is in place, you can take the command and forward it to an JS library implementation or API endpoint.

Reviewer: ```suggestion
This is a example Node implementation of a CLI wrapper for an _____ implementation. Once the CLI is in place, you can take the command and forward it to a JS library implementation or API endpoint.
```

Author: Something along these lines?

```suggestion
This is a example Node CLI wrapper that can help you integrate either:
- JS library implementation that you should add to the `dependencies` section of `package.json` and import it inside the `main.js`
- Framework agnostic server implementation that is accessible via HTTP API. You can use `fetch` to make the requests.
```

Reviewer: Getting better... What do you think of this?
```suggestion
This is an example Node CLI wrapper that can help you integrate either:
- a JavaScript library implementation that you should add to the `dependencies` section of `package.json` and import inside the `main.js`.
- a Framework-agnostic server implementation that is accessible via HTTP API. You can use `fetch` to make the requests.
```

Author: Looks good to me 5ebc557

## Running

> The wrapper exposes `validate` CLI command. If you want to try this locally outside of the Docker environment, navigate to this package `cd implementations/example-node` and run the following command:

Reviewer: ```suggestion
The wrapper exposes the `validate` CLI command. If you want to try this locally outside of the Docker environment, navigate to this package (e.g., `cd implementations/example-node`), and run the following commands:
```

```bash
npm install
npm link
```

> After this you will be able to run the implementation without the need for Docker

Reviewer: ```suggestion
Subsequently, you will be able to run the implementation without needing Docker:
```

```bash
validate --input <input-file>  --config <config-json> --output <output-file> 
```

For example:

```bash
validate \
  --input ../../tests/input/valid-credential.json \
  --config '{"check":"identifier"}' \
  --output ../../tests/output/1-sample-impl.json
```

## Docker

### Building

To build the Docker image:

```bash
docker build -t example-node-validator .
```

### Running

To run the validator using Docker:

```bash
docker run -v $(pwd)/tests:/tests example-node-validator validate \
  --input /tests/input/valid-credential.json \
  --config '{"check":"identifier"}' \
  --output /tests/output/1-sample-impl.json
```

> Make sure to mount the appropriate directories for input and output files.

Reviewer: ~~~suggestion
To run the validator using Docker, mount the appropriate directories for input and output files, and issue a command like the following:

```bash
docker run -v $(pwd)/tests:/tests example-node-validator validate \
  --input /tests/input/valid-credential.json \
  --config '{"check":"identifier"}' \
  --output /tests/output/1-sample-impl.json
```
~~~
